File: BFS_DFS/baekjoon_1260.py
from collections import deque
import sys

def graph(n, line):
    Graph = [[] for _ in range(n+1)]
    
    for j in line:
        for i in range(1, n+1):
            if j[0] == i:
                Graph[i].append(j[1])
            elif j[1] == i:
                Graph[i].append(j[0])
            else: 
                continue
         
    return [sorted(i) for i in Graph]

def dfs(Graph, v, visited):
    
    visited[v] = True
    print(v, end = ' ')
     
    for i in Graph[v]:
        if not visited[i]:
            dfs(Graph, i ,visited)
            
            
    
def bfs(Graph, start, visited):
    
    queue = deque([start])
    
    visited[start] = True
    while queue:
        v = queue.popleft()
        print(v, end = ' ')
        
        
        for i in Graph[v]:
            if not visited[i]:
                queue.append(i)
                visited[i] = True
            
    
if __name__ == '__main__':
    n,m,v = map(int, input().split())
    line = [list(map(int, sys.stdin.readline().split())) for _ in range(m)]
    Graph = graph(n,line)
    
    visited = [False] * (n+1)
    dfs(Graph, v, visited)
    print()
    
    visited = [False] * (n+1)
    bfs(Graph, v, visited)
    
    

File: BFS_DFS/baekjoon_1260.c
#include<stdio.h>

int graph[1001][1001];
int dfsvisited[1001]={0};
int bfsvisited[1001]={0};
int queue[1001];

void BFS(int v, int n){

    int front = 0;
    int rear = 0;
    int i,POP;

    queue[0] = v;
    printf("%d ", v);
    bfsvisited[v] = 1;
    rear++;

    while(front<rear){

        POP = queue[front];
        front++;

        for(i=1; i<=n; i++){
            if(graph[POP][i] == 1 && bfsvisited[i] == 0){
                queue[rear] = i;
                bfsvisited[i] = 1;
                printf("%d ", i);
                rear++;
            }

       }
    }
    return;
}

void DFS(int v, int n){

    int i;

    dfsvisited[v] = 1;
    printf("%d ", v);

    for(i=1; i<=n; i++){
        if((graph[v][i] == 1) && dfsvisited[i] == 0){
                DFS(i,n);

        }
    }
    return;
}

int main(){

    int n,m,v;
    int i,x,y;

    scanf("%d %d %d", &n, &m, &v);

    for(i=1; i<=m; i++){
        scanf("%d %d", &x, &y);
        graph[x][y] = graph[y][x] = 1;
    };


    DFS(v,n);
    printf("\n");
    BFS(v,n);

    return 0;
}


